pkg/scalers/pulsar_scaler.go

```diff
@@ -0,0 +1,313 @@
+		cert, err := tls.X509KeyPair([]byte(pulsarMetadata.cert), []byte(pulsarMetadata.key))
+		if err != nil {
+			pulsarLog.Error(err, "unable to load cert or key file")
+			return nil, fmt.Errorf("unable to load cert or key file: %s", err)
+		}
+
+		rootCAs, _ := x509.SystemCertPool()
+		if rootCAs == nil {
+			rootCAs = x509.NewCertPool()
+		}
+
+		// Append our cert to the system pool
+		if ok := rootCAs.AppendCertsFromPEM([]byte(pulsarMetadata.ca)); !ok {
+			pulsarLog.Info("No certs appended, using system certs only")
+		}
+		// Setup HTTPS client
+		tlsConfig := &tls.Config{
+			InsecureSkipVerify: false,
+			Certificates:       []tls.Certificate{cert},
+			RootCAs:            rootCAs,
+		}
+
+		transport := &http.Transport{TLSClientConfig: tlsConfig}
+		client = &http.Client{Transport: transport}
+	}
+
+	return &pulsarScaler{
+		client:   client,
+		metadata: pulsarMetadata,
+	}, nil
+}
+
+func parsePulsarMetadata(config *ScalerConfig) (pulsarMetadata, error) {
+	meta := pulsarMetadata{}
+	switch {
+	case config.TriggerMetadata["adminURLFromEnv"] != "":
+		meta.adminURL = config.ResolvedEnv[config.TriggerMetadata["adminURLFromEnv"]]
+	case config.TriggerMetadata["adminURL"] != "":
+		meta.adminURL = config.TriggerMetadata["adminURL"]
+	default:
+		return meta, errors.New("no adminURL given")
+	}
+
+	switch {
+	case config.TriggerMetadata["topicFromEnv"] != "":
+		meta.topic = config.ResolvedEnv[config.TriggerMetadata["topicFromEnv"]]
+	case config.TriggerMetadata["topic"] != "":
+		meta.topic = config.TriggerMetadata["topic"]
+	default:
+		return meta, errors.New("no topic given")
+	}
+
+	switch {
+	case config.TriggerMetadata["subscriptionFromEnv"] != "":
+		meta.subscription = config.ResolvedEnv[config.TriggerMetadata["subscriptionFromEnv"]]
+	case config.TriggerMetadata["subscription"] != "":
+		meta.subscription = config.TriggerMetadata["subscription"]
+	default:
+		return meta, errors.New("no subscription given")
+	}
+
+	meta.msgBacklogThreshold = defaultMsgBacklogThreshold
+
+	if val, ok := config.TriggerMetadata[msgBacklogMetricName]; ok {
+		t, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return meta, fmt.Errorf("error parsing %s: %s", msgBacklogMetricName, err)
+		}
+		meta.msgBacklogThreshold = t
+	}
+
+	meta.enableTLS = false
+	if val, ok := config.TriggerMetadata["tls"]; ok {
+		val = strings.TrimSpace(val)
+
+		if val == "enable" {
+			cert := config.AuthParams["cert"]
+			key := config.AuthParams["key"]
+			if key == "" || cert == "" {
+				return meta, errors.New("must be provided cert and key")
+			}
+			meta.ca = config.AuthParams["ca"]
+			meta.cert = cert
+			meta.key = key
+			meta.enableTLS = true
+		} else {
+			return meta, fmt.Errorf("err incorrect value for TLS given: %s", val)
+		}
+	}
+	return meta, nil
+}
+
+func (s *pulsarScaler) GetStats() (*pulsarStats, error) {
+	stats := new(pulsarStats)
+
+	topic := strings.ReplaceAll(s.metadata.topic, "persistent://", "")
```

Comment:
Do we need the full name topic in any case? I mean, inside the scaler the only usage I have seen is this and we replace the given value every time. Maybe if the scaler requests the topic with the prefix "persistent://" (because it's the common language in pulsar for instance) , maybe we can parse in during `parsePulsarMetadata` to do it only once.
WDYT?

Reply:
Yes, you absolutely right, we just need to parse it once.